test/openssl/test_pkey.rb

```diff
@@ -244,4 +244,210 @@ def test_to_text
     rsa = Fixtures.pkey("rsa1024")
     assert_include rsa.to_text, "publicExponent"
   end
+
+  if openssl?(3, 0, 0)
+    def test_from_parameters_with_invalid_alg
+      e = assert_raise(OpenSSL::PKey::PKeyError) {
+        OpenSSL::PKey.from_parameters("ASR", {})
+      }
+      assert_match(/^EVP_PKEY_CTX_new_from_name: unsupported/, e.message)
```

Comment:
```suggestion
      assert_raise_with_message(OpenSSL::PKey::PKeyError, /^EVP_PKEY_CTX_new_from_name: unsupported/) {
        OpenSSL::PKey.from_parameters("ASR", {})
      }
```

We can use `assert_raise_with_message` for this kind of test.